### PR TITLE
Misc cleanup in o.e.search

### PR DESCRIPTION
**Pull request by original-brownbear**
Cleaning up a few more spots here.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java [hunk 1]
```diff
@@ -174,37 +174,6 @@ public FloatVectorValues getFloatVectorValues(String field) throws IOException {
...
-
-        @Override
-        public void searchNearestVectors(String field, float[] target, KnnCollector collector, Bits acceptDocs) throws IOException {
-            if (queryCancellation.isEnabled() == false) {
-                in.searchNearestVectors(field, target, collector, acceptDocs);
-                return;
-            }
-            // when acceptDocs is null due to no doc deleted, we will instantiate a new one that would
-            // match all docs to allow timeout checking.
-            final Bits updatedAcceptDocs = acceptDocs == null ? new Bits.MatchAllBits(maxDoc()) : acceptDocs;
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java:186]: This is the only change that isn't so obvious I think. The `timeoutCheckingAcceptDocs` variable wasn't used. If you remove it, the check for `queryCancellation.isEnabled()` becomes redundant and the method collapses into what the parent method does.

File: server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java [hunk 2]
```diff
@@ -174,37 +174,6 @@ public FloatVectorValues getFloatVectorValues(String field) throws IOException {
...
-                    return updatedAcceptDocs.get(index);
-                }
-
-                @Override
-                public int length() {
-                    return updatedAcceptDocs.length();
-                }
-            };
-
-            in.searchNearestVectors(field, target, collector, acceptDocs);
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java:206]: We should not delete this. The bug is that `in.searchNearestVectors(field, target, collector, acceptDocs);` 

it should be 
```
in.searchNearestVectors(field, target, collector, timeoutCheckingAcceptDocs);
```

File: server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java [hunk 3]
```diff
@@ -240,7 +240,7 @@ public ShardSearchRequest(
         this.originalIndices = originalIndices;
         this.readerId = readerId;
         this.keepAlive = keepAlive;
-        assert keepAlive == null || readerId != null : "readerId: " + readerId + " keepAlive: " + keepAlive;
+        assert keepAlive == null || readerId != null : "readerId: null keepAlive: " + keepAlive;
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java:243]: I don't understand this.

This assertion is tripped if:
 - keepAlive is `null`
 - OR `readerId` ISN'T `null`.

Why do we replace readerId in the string builder with `null`? Is it always passed in as null in the ctor?

File: server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java [hunk 4]
```diff
@@ -334,7 +334,7 @@ public ShardSearchRequest(StreamInput in) throws IOException {
             readerId = null;
             keepAlive = null;
         }
-        assert keepAlive == null || readerId != null : "readerId: " + readerId + " keepAlive: " + keepAlive;
+        assert keepAlive == null || readerId != null : "readerId: null keepAlive: " + keepAlive;
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java:337]: same concerns as above, I don't understand this.

**Review by benwtrent — COMMENTED** [review]
some concerns, but overall looks good.

File: server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java (hunk 3 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java:243]: If keep alive is not null, then the assertion is only tripped if the reader is null.
If keep alive is null, the assertions isn't tripped.
=> reader is always null if assertion is tripped :)

File: server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java (hunk 2 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java:206]: 🤦 sorry about that, I tried to make up for my mindlessness by drying this up properly now at least :)

File: server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java [hunk 5]
```diff
@@ -183,27 +164,30 @@ public void searchNearestVectors(String field, float[] target, KnnCollector coll
...
-
-                @Override
-                public boolean get(int index) {
-                    if (calls++ % MAX_CALLS_BEFORE_QUERY_TIMEOUT_CHECK == 0) {
-                        queryCancellation.checkCancelled();
-                    }
-
-                    return updatedAcceptDocs.get(index);
-                }
+            in.searchNearestVectors(field, target, collector, acceptDocs);
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java:167]:
```suggestion
            in.searchNearestVectors(field, target, collector, new TimeOutCheckingBits(acceptDocs));
```
?

File: server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java (hunk 3 above)
Comment (benwtrent) [reply server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java:243]: @original-brownbear 🤦 yes, I was reading it backwards!

File: server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java (hunk 5 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java]: Uff sorry about that! ... fixed now, moved the comment as well.

**Review by benwtrent — APPROVED** [review]

**Comment by original-brownbear** [comment]
Thanks @benwtrent !

**Pull request merged**